phoenix-core/src/main/antlr3/PhoenixSQL.g

```diff
@@ -967,9 +967,14 @@ NUMBER
     :   POSINTEGER
     ;
 
-// Exponential format is not supported.
 DECIMAL
```

Comment:
I don't think any new methods in ParseNodeFactory will be necessary, as you can use the factory.literal(Object o) method and if the Object is an instanceof Double, it'll do the right thing. You will need the grammar change, but just create a new top level rule for it instead of adding it to DECIMAL. When it matches, you can do something like this:

    $ret = factory.literal(Double.valueOf(token.getText()));